src/app/_common/data/AdData.jsx

```diff
@@ -0,0 +1,247 @@
+        jobArrangement: getString(properties.jobarrangement),
+        jobPercentage: getJobPercentage(properties.jobpercentage),
+        jobTitle: getString(properties.jobtitle),
+        positionCount: getString(properties.positioncount),
+        remote: getString(properties.remote),
+        sector: getString(properties.sector),
+        startTime: getString(properties.starttime),
+        workday: getWorktime(properties.workday),
+        workhours: getWorktime(properties.workhours),
+        workLanguages: getArray(properties.workLanguage),
+        locationList: getLocationListData(data.locationList),
+        location: getString(properties.location),
+
+        // Employer
+        employer: getEmployerData(data),
+        contactList: getContactList(data.contactList),
+    };
+}
+
+function getAdText(adText) {
+    let processedAdText = adText;
+    if (containsEmail(adText)) {
+        try {
+            const extractedEmails = [...extractEmail(adText)];
+            let preprocessedAd = adText.replace(/&#64;/g, "@");
+            extractedEmails.forEach((it) => {
+                if (isValidEmail(it) && !mailtoInString(preprocessedAd, it)) {
+                    preprocessedAd = preprocessedAd.replace(it, `<a rel="nofollow" href="mailto:${it}">${it}</a>`);
+                }
+            });
+            processedAdText = preprocessedAd;
+        } catch (err) {
+            processedAdText = adText;
+        }
+    }
+    // TODO: double check that sanitizing doesn't remove <section> (see amplitude.js for usage)
+    return DOMPurify.sanitize(processedAdText);
+}
+
+function getEmail(email) {
+    return isValidEmail(email) ? email : undefined;
+}
+
+function getJobPercentage(jobPercentage) {
+    if (!jobPercentage) {
+        return undefined;
+    }
+
+    return jobPercentage + jobPercentage.endsWith("%") ? "" : "  %";
+}
+
+function getWorktime(worktime) {
+    if (!worktime) {
+        return undefined;
+    }
+
+    // We need this check in case of old workhour/-day property values, formatted like 'Opt1 Opt2'
+    const items = [];
+    try {
+        const jsonArray = JSON.parse(worktime);
+
+        for (let i = 0; i < jsonArray.length; i += 1) {
+            items.push(jsonArray[i]);
+        }
+    } catch (e) {
+        items.push(worktime);
+    }
+    return items.join(", ");
+}
+
+function getContactList(contactList) {
+    if (!contactList) {
+        return undefined;
+    }
+
+    return contactList.map((contact) => {
+        return {
+            name: contact.name,
+            title: contact.title,
+            phone: contact.phone,
+            email: getEmail(contact.email),
+        };
+    });
+}
+
+/**
+ *  --------------------------- Common Functions ---------------------------
+ */
+function getString(value) {
+    if (value && typeof value === "string") {
+        return value;
+    }
+    return undefined;
+}
+
+function getArray(arrayData) {
+    return Array.isArray(arrayData) ? arrayData : [];
```

Comment:
Siden `getString()` og andre hjelpemetoder returnerer `undefined` hvis datafeltet ikke finnes, burde man kanskje returnere `undefined` her også istedet for en tom array?

Reply:
enig, fixed